### DISCUSSION TITLE
Wizarr V3

### DISCUSSION THREAD

**Discussion by realashleybailey** [wizarrrr/wizarr#188] [Q&A]
# Wizarr 3.0.0

Wizarr 3.0.0 is under development with alot of Wizarr being rewritten and introducing minor design changes, I wish to open this forum so that I can compile any Q&A the community may have and also to collect feedback on bugs that have occured in Wizarr V2.

## Feature Requests
If you have requests for features that you want in V3 then feel free to leave them, however I can not promise that these will be introduced in the initial beta or release as there is already a massive undertaking to get the initial release stable enough to be brought to the master branch.

Please do not ask for multi server support, this has been requested alot and we hear you and it is something that may come in the future, but as of right now bringing support for both Plex and Jellyfin under the same instance would require massive database changes and would take a long time, we recommend running two instances of Wizarr for now.

## Bug Reports
If you have found a bug in Wizarr V2 then please report it here, I will be going through the issues and fixing them as I can for V3.

## Tracking Progress
If you wish to track the progress of the V3 in its alpha stage then it can be found under the ```ghcr.io/wizarrrr/wizarr:v3-alpha``` docker image, or at [https://github.com/Wizarrrr/wizarr/tree/v3-alpha](https://github.com/Wizarrrr/wizarr/tree/v3-alpha), we welcome PR's on the codebase if you wish to fix anything that you believe is in need of it.

## Translations
Please do not open PR's for translations on the v3-alpha as these will be ignored as translations can change alot during development (wait until v3 is stable).

## Changelog
Wizarr 3.0.0 is a major update that includes a number of new features and improvements. 

Here is a summary of the changes:
- New UI: The user interface has been slightly redesigned.
- Settings split out into sub sections: The settings page has been reorganized into a number of sub sections, making it easier to find the settings you need.
- REST API: A REST API has been added, allowing you to control Wizarr through your own software.
- Multiple admins: Multiple admins can now be added, so you don't have to share login information with everyone who needs to manage Wizarr.
- URL history support: The URL history is now kept persistent, so you can easily navigate back to pages you've visited previously.
- Homepage: A homepage has been added, which can be used to display a welcome message or other information, aswell as buttons to navigate to join or login.
- Request for Invite: Users can now request an invite to Wizarr, which you can then approve or deny.
- New setup utility: A new setup utility has been added to make it easier to get started with Wizarr.
- Scheduled tasks: Scheduled tasks can now be viewed, they run at regular intervals or can be triggered manually from the settings page.
- Adding custom HTML page: Adding a custom HTML page now uses TinyMCE, which allows you to preview and format your HTML before saving.
- Typescript and SCSS support: Wizarr now supports Typescript and SCSS, which allows for more complex and powerful JavaScript functionality.
- Footer bar: A footer bar has been added to the login and admin portal, which shows the current version of Wizarr.
- Account management page: A new account management page has been added, where you can change your password and username.
- JWT tokens: Authentication has been replaced with JWT tokens, which are more secure and efficient.
- Session page: A new session page has been added, where you can see a list of devices that have logged into your Wizarr admin account.
- API keys: API keys can now be created and revoked from the settings page.
- Custom duration and expiration input: Custom duration and expiration input is now more customizable when creating invites.
- Simplified invitations page table: The invitations page table has been simplified, with a button for viewing further information.
- Profile pictures for Jellyfin users: Profile pictures are now available for Jellyfin users, and they are cached by the hour to prevent too many requests to Plex or Jellyfin.
- Scan Users button: A Scan Users button has been added, which allows you to manually scan for Media Server users.
- Add User button: A new Add User button has been added, which allows you to create Jellyfin or Plex users without linking them to an invite.
- Logout button: A logout button has been added to the admin portal.
- Rebuilt setup page: The setup page has been rebuilt to load all steps on page load and carousel through them on button press.
- Custom redirect link: The ability to set a custom redirect link after user signup has been added.
- Multiple admin account, let others take over your Wizarr and manage it for you.
- Role based accounts, the main accounts on Wizarr will now be role based, Admin, Moderator and User, with User being an extremely limited to viewing stuff only.
- OAuth Login, be able to login without a username and password using your OAuth provider of choice, once configured you can link your oauth provider account to your Wizarr account for one click sign in.
- Automated translations, Wizarr will now be translated on the fly with each new build automatically using Google Translate.
- Change Language, a button now exists to change the language your viewing Wizarr in.
- Backup Wizarr, be able to download a Wizarr backup to your computer and then restore the backup. [Possibly add function to schedule backups too]
- New About page that will give ease of access to information on your current server and also changelogs from github.
- Live logs, view the running servers logs in realtime from Wizarr frontend.
- Fully redesigned invitation flow using websockets to feedback live information on user account creation process when joining your media server.
more to come...

## Documentation
I am currently looking for someone that is willing to collab with me on building Documentation for V3, it is an extensive process as I am hoping to build a large docs for developers to follow when they wish to contribute to Wizarr. If this is something your interested in drop a message in the Discord @ [https://discord.com/invite/wsSTsHGsqu](https://discord.com/invite/wsSTsHGsqu).

## Personal Tidbit
This update is to say the least a big undertaking and my hopes were to have this completed by early August, with intermittent issues and holdbacks and also personal life events I do not believe I could have V3 completed in a state for release by that deadline. As a developer on this project I am honoured to write code that will be used by many more people and so therefore I would never release code that I could not say is stable enough, I am therefore happy to delay the release and continue working on V3 until I am satisfied that it is stable enough to be used by the community.

## Thank You
I love Wizarr and see an amazing future for the project with it sitting right next to your other docker images.

**Comment by christianmerges** [comment]
Describe the bug
After changing the repo from latest to v3 preserving the config db, it was not possible anymore to connect to the plex server. The error in the bottom right corner says, the servername is invalid. I tried using the following as servername: http://plex:32400/

Additional context
It was possible to connect to plex by using the public fqdn hostname. Communication should be possible local by using the hostname of the other container.

**Reply by realashleybailey** [reply]
Thank you for reporting, I have looked into the issue and it relates to input validation, URL are validated as http/s://[0.0.0.0]or[anything.anything][:PORT] so hosts would require there to be a dot separator. I will fix this most likely by validating URL's without the requirement for a dot separator.

**Reply by realashleybailey** [reply]
This issue has been fixed and will be available in the next docker image build.

**Reply by Baltimorepc** [reply]
i was unable to get the Alpha V3 to work  it wont take the APP_URL but it works fine on the normal wizarrr/wizarr:latest

**Reply by realashleybailey** [reply]
You are filling in APP_URL wrong, APP_URL must match your domain perfectly, DO NOT INCLUDE http:// or https://, for example if your running locally then you may wish to set APP_URL to localhost:5690 or 127.0.0.1:5690, if your running behind a reverse proxy then you may want to set the APP_URL to wizarr.mywebsite.com. I would appreciate in future you create a new thread for your discussion instead of piggybacking of another separate discussion.

**Comment by Baltimorepc** [comment]
I tried that as well didn’t work 


Sent from Yahoo Mail for iPhone


On Wednesday, August 16, 2023, 8:44 AM, Ashley Bailey ***@***.***> wrote:




You are filling in APP_URL wrong, it must be the url without http://, so localhost:5690 or example.com

**Comment by christianmerges** [comment]
Setting up a new instance, by typing in the plex url and hitting "detect server". Notfication says "Server found", but save button disapeers.

**Reply by realashleybailey** [reply]
Hi there the newest update should address these issues

**Reply by christianmerges** [reply]
I can confirm this as working now. Thanks for fixing.

**Comment by christianmerges** [comment]
When using the Link to join on a mobile device. Could u put links to the Google Play / Apple Appstore link instead of the plex website? Because on Android it is showing download for Plex for Linux.

**Reply by realashleybailey** [reply]
Yes I certainly can, I will make sure to incorporate this when I rewrite the wizard.

**Comment by christianmerges** [comment]
In the last step of the onbording guide for invited users, the plex Server itself is linked. Here the internal link of the the docker service from the configuration is used. Can u grab the external address from plex server config or make an extra field for public url in the settings?

**Reply by realashleybailey** [reply]
Yes in the V2 dev version I have added this and it will be coming over to V3 too.

**Comment by christianmerges** [comment]
When creating a link without hiting the checkmark for "unlimited usage", the link can still be used again.

**Reply by realashleybailey** [reply]
I have yet to program the usage backend code, will be completed soon though

**Comment by christianmerges** [comment]
When creating a link with expire 1 week and duration 1 Year. After the user joins, he is shown to be expired after one week. It looks like expiry and duration is mixed up. Didn't checked yet if the user really gets deleted after one week.

**Reply by realashleybailey** [reply]
Thank you for bringing this to my attention, I will be sure to fix that :)

**Comment by christianmerges** [comment]
When opening first the list of invitations and than going to the page for creating a new one, clicking on the "advanced options" link on the page where u can create links, by clicking on that, the page with list of existing links opens. It doesn't happen, when u r clicking after reloading the page.

**Reply by realashleybailey** [reply]
This will be fixed by Wizarrs migration to Vue Framework so, in about 2 weeks itll be fixed on the main V3 beta

**Comment by christianmerges** [comment]
Adding a user to Plex home doesn't work. The user get's an error message, that the invite couldn't be done. In the plex server itself then the user shows as "managed user" instead of plex home user.

**Reply by realashleybailey** [reply]
Thank you for addressing this, I will do some debugging and see if I can reproduce the error

**Comment by nathan815** [comment]
> Adding custom HTML page: Adding a custom HTML page now uses TinyMCE, which allows you to preview and format your HTML before saving.

I don't see Custom HTML in settings, is there additional configuration needed to enable this option?
